yasa/evaluation.py

```diff
@@ -0,0 +1,649 @@
+
+        Returns
+        -------
+        ax : :py:class:`matplotlib.axes.Axes`
+            Matplotlib Axes
+
+        Examples
+        --------
+        .. plot::
+
+            >>> from yasa import simulate_hypnogram
+            >>> hyp = simulate_hypnogram(seed=7)
+            >>> ax = hyp.evaluate(hyp.simulate_similar()).plot_hypnograms()
+        """
+        assert isinstance(legend, (bool, dict)), "`legend` must be True, False, or a dictionary"
+        assert isinstance(kwargs_ref, dict), "`kwargs_ref` must be a dictionary"
+        assert isinstance(kwargs_test, dict), "`kwargs_test` must be a dictionary"
+        assert not "ax" in kwargs_ref | kwargs_test, (
+            "ax can't be supplied to `kwargs_ref` or `kwargs_test`, use the `ax` keyword instead"
+        )
+        plot_kwargs_ref = {"highlight": None, "alpha": 0.8}
+        plot_kwargs_test = {"highlight": None, "alpha": 0.8, "color": "darkcyan", "ls": "dashed"}
+        plot_kwargs_ref.update(kwargs_ref)
+        plot_kwargs_test.update(kwargs_test)
+        if ax is None:
+            ax = plt.gca()
+        self.hypno_ref.plot_hypnogram(ax=ax, **plot_kwargs_ref)
+        self.hypno_test.plot_hypnogram(ax=ax, **plot_kwargs_test)
+        if legend and "label" in plot_kwargs_ref | plot_kwargs_test:
+            if isinstance(legend, dict):
+                ax.legend(**legend)
+            else:
+                ax.legend()
+        return ax
+
+    def plot_roc(self, palette=None, ax=None, **kwargs):
+        """Plot ROC curves for each stage.
+
+        Parameters
+        ----------
+        palette : dict or None
+            If a dictionary, keys are stages and values are corresponding colors.
+        ax : :py:class:`matplotlib.axes.Axes`
+            Axis on which to draw the plot, optional.
+        **kwargs : key, value pairs
+            Additional keyword arguments are passed to the :py:func:`matplotlib.pyplot.plot` call.
+
+        Returns
+        -------
+        ax : :py:class:`matplotlib.axes.Axes`
+            Matplotlib Axes
+        """
+        raise NotImplementedError("Requires probability/confidence values.")
+
+class SleepStatsEvaluation:
+    """
+    Evaluate agreement between two measurement devices by comparing summary sleep statistics across
+    multiple participants or sessions.
+
+    For example, the reference device might be PSG and the test device might be a wearable device.
+
+    Parameters
+    ----------
+    data : :py:class:`pandas.DataFrame`
+        A pandas dataframe with sleep statistics from two different
+        devices for multiple subjects
+    reference : str
+        Name of column containing the reference device sleep statistics.
+    test : str
+        Name of column containing the test device sleep statistics.
+    subject : str
+        Name of column containing the subject ID.
+    statistic : str
+        Name of column containing the name of the sleep statistics.
```

Comment:
Is this a single column or a general prefix?

Reply:
I agree this language was confusing, but now it's outdated because this dataframe is generated during `SleepStatsEvaluation.__init__`.